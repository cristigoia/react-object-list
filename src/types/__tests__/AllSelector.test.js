import React from 'react'
import { shallow } from 'enzyme'
import {
  snapshotTest,
} from 'utils/tests'
import { AllSelector } from '../'

describe('Selector type', () => {
  it('snapshot test', () => {
    snapshotTest(<AllSelector />)
    snapshotTest(<AllSelector allSelected />)
  })
  it('makes a call to the callback function when changed', () => {
    const callback = jest.fn()
    const wrapper = shallow(<AllSelector toggleSelectAll={callback} />)
    expect(callback).not.toBeCalled()
    wrapper.find('input').simulate('change')
    expect(callback).toBeCalled()
  })
})
