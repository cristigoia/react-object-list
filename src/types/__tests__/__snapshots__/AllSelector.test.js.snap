// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selector type snapshot test 1`] = `
<input
  checked={false}
  onChange={[Function]}
  type="checkbox"
/>
`;

exports[`Selector type snapshot test 2`] = `
<input
  checked={true}
  onChange={[Function]}
  type="checkbox"
/>
`;
